src/sage/matroids/chow_ring.py

```diff
@@ -0,0 +1,322 @@
+
+    where `I_{FY}(M)` is the :class:`Feitchner-Yuzvinsky augmented Chow ring
+    ideal <sage.matroids.chow_ring_ideal.AugmentedChowRingIdeal_fy>`
+    of matroid `M`.
+
+    The *atom-free presentation* is the quotient ring
+
+    .. MATH::
+
+        A(M)_R := R[x_{F_1}, \ldots, x_{F_k}] / I_{af}(M),
+
+    where `I_{af}(M)` is the :class:`atom-free augmented Chow ring ideal
+    <sage.matroids.chow_ring_ideal.AugmentedChowRingIdeal_atom_free>`
+    of matroid `M`.
+
+    .. SEEALSO::
+
+        :mod:`sage.matroids.chow_ring_ideal`
+
+    INPUT:
+
+    - ``M`` -- matroid
+    - ``R`` -- commutative ring
+    - ``augmented`` -- boolean; when ``True``, this is the augmented
+      Chow ring and if ``False``, this is the non-augmented Chow ring
+    - ``presentation`` -- string (default: ``None``); one of the following
+      (ignored if ``augmented=False``)
+
+      * ``"fy"`` - the Feitchner-Yuzvinsky presentation
+      * ``"atom-free"`` - the atom-free presentation
+
+    REFERENCES:
+
+    - [FY2004]_
+    - [AHK2015]_
+
+    EXAMPLES::
+
+        sage: M1 = matroids.catalog.P8pp() #more examples
+        sage: ch = M1.chow_ring(QQ, False)
+        sage: ch
+        Chow ring of P8'': Matroid of rank 4 on 8 elements with 8 nonspanning circuits
+    """
+    def __init__(self, R, M, augmented, presentation=None):
+        r"""
+        Initialize ``self``.
+
+        EXAMPLES::
+
+            sage: ch = matroids.Wheel(3).chow_ring(QQ, False)
+            sage: TestSuite(ch).run()
+        """
+        self._matroid = M
+        self._augmented = augmented
+        self._presentation = presentation
+        if augmented is True:
+            if presentation == 'fy':
+                self._ideal = AugmentedChowRingIdeal_fy(M, R)
+            elif presentation == 'atom-free':
+                self._ideal = AugmentedChowRingIdeal_atom_free(M, R)
+        else:
+            self._ideal = ChowRingIdeal_nonaug(M, R)
+        C = CommutativeRings().Quotients() & GradedAlgebrasWithBasis(R).FiniteDimensional()
+        QuotientRing_generic.__init__(self, R=self._ideal.ring(),
+                                      I=self._ideal,
+                                      names=self._ideal.ring().variable_names(),
+                                      category=C)
+
+    def _repr_(self):
+        r"""
+        EXAMPLES::
+
+            sage: M1 = matroids.catalog.Fano()
+            sage: ch = M1.chow_ring(QQ, False)
+            sage: ch
+            Chow ring of Fano: Binary matroid of rank 3 on 7 elements, type (3, 0)
+        """
+        if self._augmented is True:
+            if self._presentation == 'fy':
+                return "Augmented Chow ring of {} in Feitchner-Yuzvinsky presentation".format(self._matroid)
+            elif self._presentation == 'atom-free':
+                return "Augmented Chow ring of {} in atom-free presentation".format(self._matroid)
+        return "Chow ring of {}".format(self._matroid)
```

Comment:
```suggestion
        output = “Chow ring of {}”.format(self._matroid)
        if self._augmented is True:
            output = *Augmented “ + output
            if self._presentation == 'fy':
                output += “Feitchner-Yuzvinsky presentation"
            elif self._presentation == 'atom-free':
                output += “atom-free presentation"
        return output + “ over “ + repr(self.base_ring())
```
Sorry I missed this before. It would be good to include the base ring in the print repr.